cmd/metal-api/internal/fsm/events.go

```diff
@@ -159,5 +159,8 @@ func eventCallbacks(config *states.StateConfig) fsm.Callbacks {
 		callbacks["enter_"+name] = state.OnTransition
 	}
 
+	waiting := allStates[states.Waiting.String()].(*states.WaitingState)
```

Comment:
Maybe a little cleaner and more extendable for the future would be to extend the `FSMState` interface with a function `OnLeave(e *fsm.Event)`. Then we can add the callback inside the loop for all states:

```golang
	for name, state := range allStates {
		callbacks["enter_"+name] = state.OnTransition
		callbacks["leave_"+name] = state.OnLeave
	}
```

In order not having to implement the function for all states, we can also consider introducing a small helper struct:

```
type noopState struct{}

func (_ noopState) OnLeave(e *fsm.Event)      {}
func (_ noopState) OnTransition(e *fsm.Event) {}
``` 

Which we can then add to all other states (except Waiting of course):

```
type AliveState struct {
	noopState
	log       *zap.SugaredLogger
	container *metal.ProvisioningEventContainer
	event     *metal.ProvisioningEvent
}
``` 

This way, we can override whenever we want and also add other callbacks in the same way in the future.

Reply:
Cool, so now we can remove these two lines, right?